packages/client/src/components/RoomTable.tsx

```diff
@@ -1,31 +1,30 @@
-import { useEffect, useState } from 'react';
-import { Link, Heading, Box, Flex, Card, Button } from 'theme-ui';
+import { useEffect } from 'react';
+import { Link, Heading, Box, Flex, Card, Button, Grid } from 'theme-ui';
 import { Room, RoomEvent } from 'shared';
 import { gameIO } from '../lib/socket';
-import { getRooms } from '../lib/resources';
-import { post } from '../utils/rest';
+import { getRooms, deleteRoom } from '../lib/resources';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
 
 export const RoomTable = ({ username }: { username: string }) => {
-  const [rooms, setRooms] = useState<Room[]>([]);
+  const queryClient = useQueryClient();
 
-  useEffect(() => {
-    getRooms().then((data) => {
-      setRooms(data);
-    });
+  const { data, isLoading } = useQuery<Room[]>('rooms', getRooms);
+
+  const { mutate } = useMutation(deleteRoom, {
```

Comment:
Should we map these to something that is more descriptive? `mutate` is not necessarily delete... Same with data, it's quite generic, no? Don't care that much, just asking.

Reply:
those are the name of the parameters returned from the lib we can rename them sure but since you have just one of each I think they can stay there. Think at a mutation like an action that update a db where a query (data) is something returned from db. Not a big deal to rename, just the need to write more

Reply:
I understand what `mutate` means, but it's really not descriptive of what it does. If you had an update action for example, you would have to rename. But I don't mind anyway, it's fine for now.

Reply:
I think we can rename it when we will have another mutation going on, right now with one it's fine